src/pages/index.js

```diff
@@ -1,67 +1,263 @@
+// Editing UserInfo Popup
+// -------------------------------------------------------------------------------------
+const profileEditPopup = new PopupWithForm("#profile-edit-modal", {
+  handleFormSubmit: (formData) => {
+    profileEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editProfileData(formData)
+      .then((data) => {
+        userInfo.setUserInfo({
+          name: data.name,
+          about: data.about,
+        });
+        profileEditPopup.close();
+      })
+      .catch((error) => {
+        console.log("Error updating Profile:", error);
+      })
+      .finally(() => {
+        profileEditPopup.toggleUploadIndicator(false);
+      });
   },
 });
+
 profileEditPopup.setEventListeners();
 
+// -------------------------------------------------------------------------------------
+// User Avatar Instance
+// -------------------------------------------------------------------------------------
+const userAvatar = new UserAvatar({
+  avatarSelector: ".profile__image",
+});
+
+// -------------------------------------------------------------------------------------
+// Inital User Avatar Population
+// -------------------------------------------------------------------------------------
+api
+  .getUserAvatar()
+  .then((data) => {
+    userAvatar.setUserAvatar({
+      avatar: data.avatar,
+    });
+  })
+  .catch((error) => {
+    console.error("Failed to fetch profile Avatar:", error);
+  });
+
+// -------------------------------------------------------------------------------------
+// Editing UserAvatar Popup
+// -------------------------------------------------------------------------------------
+const avatarEditPopup = new PopupWithForm("#avatar-edit-modal", {
+  handleFormSubmit: (formData) => {
+    const avatarURL = formData["Image URL"];
+    avatarEditPopup.toggleUploadIndicator(true);
+
+    api
+      .editUserAvatar({ avatar: avatarURL })
+      .then((data) => {
+        userAvatar.setUserAvatar({
+          avatar: data.avatar,
+        });
+        avatarEditPopup.close();
+      })
+      .catch((error) => {
+        console.log("Error updating Avatar:", error);
+      })
+      .finally(() => {
+        avatarEditPopup.toggleUploadIndicator(false);
+      });
+  },
+});
+
+avatarEditPopup.setEventListeners();
+
+// -------------------------------------------------------------------------------------
+// Card Section Instance
+// -------------------------------------------------------------------------------------
+const cardSection = new Section(
+  {
+    items: [],
+    renderer: (cardData) => {
+      const card = new Card(
+        cardData,
+        cardTemplate,
+        (cardData) => imagePopup.open(cardData),
+        handleTrashClick,
+        updateLike
+      ).getView();
+      if (cardData.isLiked) {
+        card
+          .querySelector(".card__like-button")
+          .classList.add("card__like-button_active");
```

Comment:
Now only class `Card` knows every **card selector**, so you need to encapsulate the likes logics inside its methods and call them in `index.js`. It’s a violation of **Single responsibility** principle